Multi-part form example

Rnhatch opened the pull request:
New sample page supporting a blog post

itmightbecody commented on Skuid_Techniques/Multi_Part_Form/README.md:
```diff
@@ -0,0 +1,33 @@
+# Multi-part Form Navigation and Real-time Completion Status
+
+Skuid can be used to create very effective complex data-entry forms.  There are two principles that can help users work through an extensive data-entry process:  
```
```suggestion
Skuid can be used to create very effective and complex data entry forms.  There are two principles that can help users work through an extensive data-entry process:  
```

A suggestion I'm making throughout: the more common form of this type of flow is called `data entry form` as opposed to the hyphenated `data-entry form`. My brain had to do a double take, but [the Ngram does not lie](https://books.google.com/ngrams/graph?content=data-entry+form%2C+data+entry+form&year_start=1800&year_end=2019&corpus=26&smoothing=3&direct_url=t1%3B%2Cdata%20-%20entry%20form%3B%2Cc0%3B.t1%3B%2Cdata%20entry%20form%3B%2Cc0#t1%3B%2Cdata%20-%20entry%20form%3B%2Cc0%3B.t1%3B%2Cdata%20entry%20form%3B%2Cc0). For the sake of discoverability, let's go with the more common form.

itmightbecody commented on Skuid_Techniques/Multi_Part_Form/README.md:
```diff
@@ -0,0 +1,33 @@
+- Break up the form into chunks.
+- Provide wayfinding and progress status information. 
+
+The example below uses the Skuid wizard component to break up a form into several different sections, but also uses the vertical navigation component - with selected-state conditions - to provide more effective wayfinding and status information for the whole form.  The example shows how to integrate the vertical navigation items and the wizard buttons so that both components stay in synch with us ...
```
```suggestion
The example below uses the Wizard component to break up a form into several different sections, but also uses the Vertical Navigation component - with selected-state conditions - to provide more effective wayfinding and status information for the whole form.  The example shows how to integrate the Vertical Navigation items and the Wizard buttons so that both components stay in sync with user action. 
```

itmightbecody commented on Skuid_Techniques/Multi_Part_Form/README.md:
```diff
@@ -0,0 +1,33 @@
+
+## Notes
+
+Synching the wizard and the vertical navigation selected state:
```
```suggestion
Synching the Wizard and the Vertical Navigation selected state:
```

itmightbecody commented on Skuid_Techniques/Multi_Part_Form/README.md:
```diff
@@ -0,0 +1,33 @@
+
+Synching the wizard and the vertical navigation selected state:
+
+- A UI Only Field is used to manage selected state in the vertical navigation component.   (See the `NavPrimary` field in the `UI Control` Model).   All navigation items and wizard buttons have action steps that update this field to an appropriate value. 
```
```suggestion
- A UI-only feld is used to manage selected state in the Vertical Navigation component.   (See the `NavPrimary` field in the `UI Control` Model).   All navigation items and Wizard buttons have action steps that update this field to an appropriate value. 
```

Capitalization

itmightbecody commented on Skuid_Techniques/Multi_Part_Form/README.md:
```diff
@@ -0,0 +1,33 @@
+Synching the wizard and the vertical navigation selected state:
+
+- A UI Only Field is used to manage selected state in the vertical navigation component.   (See the `NavPrimary` field in the `UI Control` Model).   All navigation items and wizard buttons have action steps that update this field to an appropriate value. 
+- The Navigation Items and the Wizard buttons each have an wizard component action that moves the user to the correct step. 
```
```suggestion
- The navigation items and the Wizard buttons each have a Wizard component action that moves the user to the correct step. 
```

itmightbecody commented on Skuid_Techniques/Multi_Part_Form/README.md:
```diff
@@ -0,0 +1,33 @@
+
+Including dynamic information in navigation step: 
+
+- The vertical navigation component will allow html in item labels.  This allows for formatting flexibility,  as well as the inclusion of merge syntax - to push model data into the label. 
```
```suggestion
- The Vertical Navigation component will allow HTML in item labels.  This allows for formatting flexibility, as well as the inclusion of merge syntax to push model data into the label. 
```

itmightbecody commented on Skuid_Techniques/Multi_Part_Form/README.md:
```diff
@@ -0,0 +1,33 @@
+Including dynamic information in navigation step: 
+
+- The vertical navigation component will allow html in item labels.  This allows for formatting flexibility,  as well as the inclusion of merge syntax - to push model data into the label. 
+- Because the component is not connected to a particular model - this has to be global merge syntax  - like `{{$Model.FormDetail.data.0.status1}}`.
```
```suggestion
- Because the component is not connected to a particular model this has to be global merge syntax, like `{{$Model.FormDetail.data.0.status1}}`.
```

The global merge note is less of an aside than the dashes would suggest. Let's use commas instead.

itmightbecody commented on Skuid_Techniques/Multi_Part_Form/README.md:
```diff
@@ -0,0 +1,33 @@
+
+- The vertical navigation component will allow html in item labels.  This allows for formatting flexibility,  as well as the inclusion of merge syntax - to push model data into the label. 
+- Because the component is not connected to a particular model - this has to be global merge syntax  - like `{{$Model.FormDetail.data.0.status1}}`.
+- In order for the display within the navigation item to be updated when the data is updated - you have to call a Javscript snippet when the data is updated.   Look at the model action on the UI Control model (where the status fields are located),  as well as the Javascript found on the page. 
```
```suggestion
- In order for the display within the navigation item to be updated when the data is updated you have to call a JavaScript snippet when the data is updated.  Look at the model action on the UI Control model (where the status fields are located),  as well as the JavaScript found on the page. 
```

itmightbecody reviewed (changes requested):
Left my general copy editing notes on the README, but also have some UX things:

**Actual bug**:
- Each **Related item** header does not properly update for their appropriate section. For example, if you add 3 products, then click to assets or contacts, the header will say "3 assets added" or "3 contacts added," even if none are there.



**Minor notes that I leave to your discretion**:
- It feels a little odd to have the headers be smaller than the navigation item titles for the side nav
- First 3 options on Options tab cannot be unselected, but the last can.
- The Email Opt Out does not update the questions complete header if you unselect it (Which I mean, _could_ be an intentional choice the user is making on the form. So this one isn't super important.)

Feel free to change these or leave them as-is.

Rnhatch commented:
@itmightbecody  I made 2 changes to the skuid page in light of your comments.  
1. Fixed the reference in wizard step header to retrieve the correct number of rows created for the  assets and contacts related items.

2. Adjusted the weight of the vertical navigation labels. 

The discrepancy in selecting and unselecting is because the first fields are picklists and the last one is a multipicklist.  I've also filed a product bug that "button list" renderer for picklists does not support the "Display None option" property.   

Thanks again for your review.  Let me know if you have further questions.

itmightbecody commented on Skuid_Techniques/Multi_Part_Form/Multi_Part_Form.xml:
```diff
@@ -0,0 +1,777 @@
+									<components>
+										<skuid__wrapper uniqueid="sk-G0t-28420" heightStrategy="fixed" height="60">
+											<components>
+												<skuid__text contents="Step 5.&amp;nbsp; &amp;nbsp;{{Sec5Summary}} Assets added.&amp;nbsp;&amp;nbsp;" uniqueid="sk-G0t-28421" styleSettingsVariant="increased" model="UI_Control">
```
```suggestion
												<skuid__text contents="Step 5.&amp;nbsp; &amp;nbsp;{{Sec5Summary}} assets added.&amp;nbsp;&amp;nbsp;" uniqueid="sk-G0t-28421" styleSettingsVariant="increased" model="UI_Control">
```

One last capitalization thing. Gotta make it nice since it's in SamplePages!

itmightbecody reviewed (approved):
Nice! Thanks for addressing everything. I've  left one final, _very_ minor copy note. I'm gonna go ahead and mark this as approved so you can implement that change and merge whenever works best for you.

Thank you Rob!
